### PR TITLE
[spirv] Add OpLine for function calls

### PR DESCRIPTION
**Pull request by antiagainst**
This covers intrinsic functions, intrinsic methods, and user-
defined functions.

### CODE REVIEW

File: tools/clang/test/CodeGenSPIRV/spirv.debug.opline.hlsl [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  return val;
+}
+
+// Note that we do two passes for debuging info: first preprocessing and then
+// compiling the preprocessed source code.
+// Because the preprocessor prepends a "#line 1 ..." line to the whole file,
+// the compliation sees line numbers incremented by 1.
+
+float4 main(uint val : A) : SV_Target {
+  // CHECK:      OpLine [[file]] 23 12
```
Comment (ehsannas) [inline tools/clang/test/CodeGenSPIRV/spirv.debug.opline.hlsl]: Shouldn't this be line 22?

File: tools/clang/test/CodeGenSPIRV/spirv.debug.opline.hlsl [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  // CHECK-NEXT: OpLoad %uint %val
+  // CHECK-NEXT: OpBitReverse
+  uint a = reversebits(val);
+
+  // CHECK:      OpLine [[file]] 27 12
```
Comment (ehsannas) [inline tools/clang/test/CodeGenSPIRV/spirv.debug.opline.hlsl]: Shouldn't this be line 26?

File: tools/clang/test/CodeGenSPIRV/spirv.debug.opline.hlsl [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  // CHECK-NEXT: OpLoad %uint %a
+  uint b = foo(a);
+
+  // CHECK:      OpLine [[file]] 31 14
```
Comment (ehsannas) [inline tools/clang/test/CodeGenSPIRV/spirv.debug.opline.hlsl]: Shouldn't this be line 30?

File: tools/clang/test/CodeGenSPIRV/spirv.debug.opline.hlsl (hunk 1 above)
Comment (antiagainst) [reply tools/clang/test/CodeGenSPIRV/spirv.debug.opline.hlsl]: It's tricky. See the comment in the above.

Comment (antiagainst) [reply tools/clang/test/CodeGenSPIRV/spirv.debug.opline.hlsl]: We are not using the exact original file in the compilation, but a preprocessed version of it.

Comment (ehsannas) [reply tools/clang/test/CodeGenSPIRV/spirv.debug.opline.hlsl]: Ah Sorry I don't know why i didn't see the comment before. Thanks

Comment (antiagainst) [reply tools/clang/test/CodeGenSPIRV/spirv.debug.opline.hlsl]: You can try to do `-P` over a normal HLSL source. If it does not include other files, there is one `#line 1 ...` prepended to it.

**Comment by antiagainst** [comment]
Apparently the current CodeGen infra is not very friendly for debug info. Quite difficult to get debug line info emitted for all instructions. So we may need to wait upon the completion of the refactoring to add support for all instructions.

**Comment by ehsannas** [comment]
SG

**Pull request merged**